### PR TITLE
Bugfix for term.split_seqs('term.right(3)') #197

### PR DESCRIPTION
**Pull request by jquast**
In version 1.18.0 and earlier,

       >> term.split_seqs(term.move_right(333) + 'xyz', maxsplit=1)
       ['\x1b[333C', '333', 'xyz']

This is a bug (#197) it duplicates the matched parameter, this is now corrected:

       ['\x1b[3C', 'xyz']

Also,
- fix flake8 linting of tests by moving fixtures to conftest.py,
  fixes "unused" or "re-definition from import" errors.
- version stamp blessed/__init__.py like a codegen step i guess
- remove run_codecov.py, its been fixed upstream
  https://github.com/codecov/codecov-python/issues/158#issuecomment-737017579

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -38,7 +38,6 @@ matrix:
       env: PATH=/c/Python38:/c/Python38/Scripts:$PATH TOXENV=py38,codecov COVERAGE_ID=travis-ci TEST_KEYBOARD=no
 
   allow_failures:
-    - python: 3.9-dev
```
Comment (avylove) [inline .travis.yml:41]: Also need to change 3.9-dev to 3.9 above

File: blessed/terminal.py [hunk 2]
```diff
@@ -1100,20 +1100,36 @@ def strip_seqs(self, text):
         """
         return Sequence(text, self).strip_seqs()
 
-    def split_seqs(self, text, **kwds):
+    def split_seqs(self, text, maxsplit=0, flags=0):
```
Comment (avylove) [inline blessed/terminal.py:1103]: I don't think flags can work here, so we may want to drop it. I get this error:
`ValueError: cannot process flags argument with a compiled pattern`

File: tox.ini [hunk 3]
```diff
@@ -208,10 +212,8 @@ commands = {envbindir}/sphinx-build -v -W -d {toxinidir}/docs/_build/doctrees -b
 [testenv:codecov]
 basepython = python{env:TOXPYTHON:{env:TRAVIS_PYTHON_VERSION:3.8}}
 passenv = TOXENV CI TRAVIS TRAVIS_* CODECOV_*
-deps = codecov>=1.4.0
-# commands = codecov -e TOXENV
-# Workaround for https://github.com/codecov/codecov-python/issues/158
-commands = {envpython} run_codecov.py -e TOXENV
```
Comment (avylove) [inline tox.ini:214]: This is awesome if it's really fixed.

File: version.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
```
Comment (avylove) [inline version.py:1]: Probably better than maintaining it in two places, but what is version.json used for? I typically set __version__ in __init__.py and parse it in setup.py.

File: tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# std imports
+import os
+
+# 3rd party
+import pytest
+
+all_terms_params = 'xterm screen ansi vt220 rxvt cons25 linux'.split()
```
Comment (avylove) [inline tests/conftest.py:7]:
```
if platform.system() == 'Windows':
    all_terms_params = ['vtwin10', ]
```

File: version.py (hunk 4 above)
Comment (jquast) [reply version.py:1]: https://github.com/jquast/blessed/blob/master/docs/conf.py#L102-L105

File: tox.ini (hunk 3 above)
Comment (jquast) [reply tox.ini:214]: yeah.. travis-ci has gone to shit and will soon be charging us, so there are other files to write now, for github actions, i suppose

File: version.py (hunk 4 above)
Comment (avylove) [reply version.py:1]: Here's what I do for enlighten. Same for most of the packages I maintain.
https://github.com/Rockhopper-Technologies/enlighten/blob/master/setup_helpers.py#L21-L32
https://github.com/Rockhopper-Technologies/enlighten/blob/master/doc/conf.py#L64
https://github.com/Rockhopper-Technologies/enlighten/blob/master/setup.py#L29

File: tox.ini (hunk 3 above)
Comment (avylove) [reply tox.ini:214]: It's been on my todo list to start looking at github actions for the same reason. Haven't gotten around to it yet.

**Comment by avylove** [comment]
I guess this needs to be updated too
https://github.com/jquast/blessed/blob/master/blessed/terminal.pyi#L91

**Comment by haliphax** [comment]
Going to pull this down and give it a shot this evening. Thanks! :metal:

**Comment by jquast** [comment]
> I guess this needs to be updated too
> https://github.com/jquast/blessed/blob/master/blessed/terminal.pyi#L91

I did https://github.com/jquast/blessed/pull/199/files#diff-4aa4f6315c033030489d0ebc7d7fc9a68da9e30d70d2b2751cec33141035d612L91

**Review by haliphax — APPROVED** [review]
Confirmed!

**Comment by avylove** [comment]
> > I guess this needs to be updated too
> > https://github.com/jquast/blessed/blob/master/blessed/terminal.pyi#L91
> 
> I did https://github.com/jquast/blessed/pull/199/files#diff-4aa4f6315c033030489d0ebc7d7fc9a68da9e30d70d2b2751cec33141035d612L91

Oh, sorry. I missed that.

**Pull request merged**